### PR TITLE
Add information on supported compiler versions

### PR DESCRIPTION
**Pull request by julianmenzler**

### CODE REVIEW

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -14,7 +14,7 @@
 | libclang-dev     | 7.1              |    Linux |                             Yes (JIT) |
 | libnuma-dev      | any              |    Linux |                            Yes (numa) |
 | libnuma1         | any              |    Linux |                            Yes (numa) |
-| llvm             | any              |    All   |                 Yes (code sanitizers) |
+| llvm             | <= 7.0.1         |    All   |                 Yes (code sanitizers) |
```
Comment (mrks) [inline DEPENDENCIES.md]: <= ?

File: README.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ Subsequent calls to CMake, e.g., when adding files to the build will not be nece
 
 ### Compiler choice
 CMake will default to your system's default compiler.
-To use a different one, call like `cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..` in a clean build directory.
+To use a different one, call like `cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ ..` in a clean build directory. See [dependencies](DEPENDENCIES.md) for supported compiler versions.
```
Comment (mrks) [inline README.md]: Not your change, but where does the "like" come from?

File: DEPENDENCIES.md (hunk 1 above)
Comment (julianmenzler) [reply DEPENDENCIES.md]: Support up to version `7.0.1`. 
Is there also a minimum version?

File: README.md (hunk 2 above)
Comment (julianmenzler) [reply README.md]: True, makes no sense.

File: DEPENDENCIES.md (hunk 1 above)
Comment (julianmenzler) [reply DEPENDENCIES.md]: Oh! I wanted to refer to the compiler frontend, which is `clang`, not `llvm`.  I'll remove this change.

Comment (mrks) [reply DEPENDENCIES.md]: Ah, for JIT, this makes sense.

**Pull request merged**